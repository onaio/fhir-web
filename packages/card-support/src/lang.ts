import i18n, { namespace } from './mls';
import { Dictionary } from '@onaio/utils';

export type Lang = Dictionary<string>;
const lang: Lang = {};

/** recompute values */
function fill() {
  lang.ERROR_OCCURRED = i18n.t(`An error occurred`, { ns: namespace });
  lang.DOWNLOAD_CLIENT_DATA = i18n.t(`Download Client Data`, { ns: namespace });
  lang.CLIENT_LOCATION = i18n.t(`Client Location`, { ns: namespace });
  lang.ALL_LOCATIONS = i18n.t(`All Locations`, { ns: namespace });
  lang.CARD_STATUS = i18n.t(`Card Status`, { ns: namespace });
  lang.NEEDS_CARD = i18n.t(`Needs Card`, { ns: namespace });
  lang.CARD_NOT_NEEDED = i18n.t(`Card not needed`, { ns: namespace });
  lang.BOTH_NEEDS_CARD_CARD_NOT_NEEDED = i18n.t(`Both "Needs card" and "Card not needed"`, {
    ns: namespace,
  });
  lang.CARD_ORDER_DATE = i18n.t(`Card Order Date`, { ns: namespace });
  lang.CARD_ORDER_DATE_REQUIRED = i18n.t(`Please enter start date and end date`, { ns: namespace });
  lang.SELECT_CARD_ORDER_DATE = i18n.t(`Select Card Order Date`, { ns: namespace });
  lang.DOWNLOADING = i18n.t(`Downloading`, { ns: namespace });
  lang.DOWNLOAD_CSV = i18n.t(`Download CSV`, { ns: namespace });
  lang.NO_DATA_FOUND = i18n.t(`No data found`, { ns: namespace });
}

// run it initial
fill();

// bind some events and fill values again (doing the magic you expect to happen magically)
i18n.on(`languageChanged`, () => {
  fill();
});

// export the const
export default lang;
