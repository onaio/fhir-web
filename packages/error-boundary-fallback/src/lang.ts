import i18n, { namespace } from './mls';
import { Dictionary } from '@onaio/utils';

const lang: Dictionary<string> = {};

/** recompute values */
function fill() {
  lang.PAGE_TITLE = i18n.t(`An Error Occurred`, { ns: namespace });
  lang.PAGE_SUB_TITLE = i18n.t(
    `There has been an error. Itâ€™s been reported to the site administrators via email and should be fixed shortly. Thanks for your patience.'`,
    { ns: namespace }
  );
  lang.BUTTON_TITLE = i18n.t(`Back Home`, { ns: namespace });
}

// run it initial
fill();

// bind some events and fill values again (doing the magic you expect to happen magically)
i18n.on(`languageChanged`, () => {
  fill();
});

// export the const
export default lang;
