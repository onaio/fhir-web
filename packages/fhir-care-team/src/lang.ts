import i18n, { namespace } from './mls';
import { Dictionary } from '@onaio/utils';

export type Lang = Dictionary<string>;

const lang: Lang = {};

/** recompute values */
function fill() {
  // Tooltips
  lang.TIP_REQUIRED_FIELD = i18n.t(`This is a required field`, { ns: namespace });

  // Errors
  lang.ERROR_OCCURRED = i18n.t(`An error occurred`, { ns: namespace });

  // Messages
  lang.CARE_TEAMS_UPDATE_SUCCESS = i18n.t(`Successfully Updated Care Teams`, { ns: namespace });
  lang.CARE_TEAMS_ADD_SUCCESS = i18n.t(`Successfully Added Care Teams`, { ns: namespace });
  lang.CARE_TEAM_DELETE_SUCCESS = i18n.t(`Successfully Deleted Care Team`, { ns: namespace });
  lang.CONFIRM_DELETE = i18n.t(`Are you sure you want to delete this Care Team?`, {
    ns: namespace,
  });
  lang.YES = i18n.t(`Yes`, { ns: namespace });
  lang.NO = i18n.t(`No`, { ns: namespace });

  // Rendered text
  lang.STATUS = i18n.t(`Status`, { ns: namespace });
  lang.NAME = i18n.t(`Name`, { ns: namespace });
  lang.SAVING = i18n.t(`Saving`, { ns: namespace });
  lang.SAVE = i18n.t(`Save`, { ns: namespace });
  lang.ACTIVE = i18n.t(`Active`, { ns: namespace });
  lang.INACTIVE = i18n.t(`Inactive`, { ns: namespace });
  lang.CANCEL = i18n.t(`Cancel`, { ns: namespace });
  lang.EDIT_CARE_TEAM = i18n.t(`Edit Care Team`, { ns: namespace });
  lang.CREATE_CARE_TEAM = i18n.t(`Create Care Team`, { ns: namespace });
  lang.EDIT = i18n.t(`Edit`, { ns: namespace });
  lang.CREATE = i18n.t(`Create`, { ns: namespace });
  lang.IDENTIFIER = i18n.t(`Identifier`, { ns: namespace });
  lang.TEAMS = i18n.t(`Care Teams`, { ns: namespace });
  lang.CARE_TEAM_MEMBERS = i18n.t(`Care Teams`, { ns: namespace });
  lang.SEARCH = i18n.t(`Search`, { ns: namespace });
  lang.IDENTIFIER = i18n.t(`Identifier`, { ns: namespace });
  lang.CARE_TEAM_PAGE_HEADER = i18n.t(`FHIR Care Team`, { ns: namespace });
  lang.SUBJECT = i18n.t(`Subject`, { ns: namespace });
  lang.PARTICIPANTS = i18n.t(`Participant`, { ns: namespace });
}

// run it initial
fill();

// bind some events and fill values again (doing the magic you expect to happen magically)
i18n.on(`languageChanged`, () => {
  fill();
});

// export the const
export default lang;
