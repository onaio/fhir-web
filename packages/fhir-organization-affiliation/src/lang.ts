import i18n, { namespace } from './mls';
import { Dictionary } from '@onaio/utils';

export type Lang = Dictionary<string>;

const lang: Lang = {};

/** recompute values */
function fill() {
  // Tooltips
  lang.TIP_REQUIRED_FIELD = i18n.t(`This is a required field`, { ns: namespace });

  // Errors
  lang.ERROR_OCCURRED = i18n.t(`An error occurred`, { ns: namespace });

  // Messages
  lang.ORG_AFFILIATION_UPDATE_SUCCESS = i18n.t(`Successfully Updated Organization Affiliation`, {
    ns: namespace,
  });
  lang.ORG_AFFILIATION_ADD_SUCCESS = i18n.t(`Successfully Added Organization Affiliation`, {
    ns: namespace,
  });
  lang.ORG_AFFILIATION_DELETE_SUCCESS = i18n.t(`Successfully Deleted Organization Affiliation`, {
    ns: namespace,
  });
  lang.CONFIRM_DELETE = i18n.t(`Are you sure you want to delete this Organization Affiliation?`, {
    ns: namespace,
  });
  lang.YES = i18n.t(`Yes`, { ns: namespace });
  lang.NO = i18n.t(`No`, { ns: namespace });

  // Rendered text
  lang.STATUS = i18n.t(`Status`, { ns: namespace });
  lang.NAME = i18n.t(`Name`, { ns: namespace });
  lang.LOCATION_NAME = i18n.t(`Practitioner Name`, { ns: namespace });
  lang.ORGANIZATION_NAME = i18n.t(`Organization Name`, { ns: namespace });
  lang.SAVING = i18n.t(`Saving`, { ns: namespace });
  lang.SAVE = i18n.t(`Save`, { ns: namespace });
  lang.ACTIVE = i18n.t(`Active`, { ns: namespace });
  lang.INACTIVE = i18n.t(`Inactive`, { ns: namespace });
  lang.CANCEL = i18n.t(`Cancel`, { ns: namespace });
  lang.EDIT_ORG_AFFILIATION = i18n.t(`Edit Organization Affiliation`, { ns: namespace });
  lang.CREATE_ORG_AFFILIATION = i18n.t(`Create Organization Affiliation`, { ns: namespace });
  lang.EDIT = i18n.t(`Edit`, { ns: namespace });
  lang.CREATE = i18n.t(`Create`, { ns: namespace });
  lang.IDENTIFIER = i18n.t(`Identifier`, { ns: namespace });
  lang.ORGANIZATION_AFFILIATION = i18n.t(`Practitioner Roles`, { ns: namespace });
  lang.SEARCH = i18n.t(`Search`, { ns: namespace });
  lang.IDENTIFIER = i18n.t(`Identifier`, { ns: namespace });
  lang.ORG_AFFILIATION_PAGE_HEADER = i18n.t(`FHIR Organization Affiliation`, { ns: namespace });
  lang.ORGANIZATION = i18n.t(`Organization`, { ns: namespace });
  lang.LOCATION = i18n.t(`Practitioner`, { ns: namespace });
}

// run it initial
fill();

// bind some events and fill values again (doing the magic you expect to happen magically)
i18n.on(`languageChanged`, () => {
  fill();
});

// export the const
export default lang;
