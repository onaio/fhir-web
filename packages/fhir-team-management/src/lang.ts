import i18n, { namespace } from './mls';
import { Dictionary } from '@onaio/utils';

export type Lang = Dictionary<string>;

const lang: Lang = {};

/** recompute values */
function fill() {
  // Tooltips
  lang.TIP_REQUIRED_FIELD = i18n.t(`This is a required field`, { ns: namespace });

  // Errors
  lang.ERROR_OCCURRED = i18n.t(`An error occurred`, { ns: namespace });

  // Messages
  lang.MSG_ASSIGN_PRACTITIONERS = i18n.t(`Assigning Practitioners`, { ns: namespace });
  lang.MSG_ASSIGN_PRACTITONERS_SUCCESS = i18n.t(`Successfully Assigned Practitioners`, {
    ns: namespace,
  });
  lang.MSG_TEAMS_UPDATE_SUCCESS = i18n.t(`Successfully Updated Teams`, { ns: namespace });
  lang.MSG_TEAMS_ADD_SUCCESS = i18n.t(`Successfully Added Teams`, { ns: namespace });

  // Rendered text
  lang.ENTER_TEAM_NAME = i18n.t(`Enter a team name`, { ns: namespace });
  lang.TEAM_NAME = i18n.t(`Team Name`, { ns: namespace });
  lang.STATUS = i18n.t(`Status`, { ns: namespace });
  lang.TEAM_MEMBERS = i18n.t(`Team Members`, { ns: namespace });
  lang.SAVING = i18n.t(`Saving`, { ns: namespace });
  lang.SAVE = i18n.t(`Save`, { ns: namespace });
  lang.ACTIVE = i18n.t(`Active`, { ns: namespace });
  lang.INACTIVE = i18n.t(`Inactive`, { ns: namespace });
  lang.SELECT_PRACTITIONER = i18n.t(`Select user (practitioners only)`, { ns: namespace });
  lang.CANCEL = i18n.t(`Cancel`, { ns: namespace });
  lang.EDIT_TEAM = i18n.t(`Edit Team`, { ns: namespace });
  lang.CREATE_TEAM = i18n.t(`Create Team`, { ns: namespace });
  lang.EDIT = i18n.t(`Edit`, { ns: namespace });
  lang.CREATE = i18n.t(`Create`, { ns: namespace });
  lang.IDENTIFIER = i18n.t(`Identifier`, { ns: namespace });
  lang.NO_TEAM_MEMBERS = i18n.t(`No team members`, { ns: namespace });
  lang.TEAMS = i18n.t(`Teams`, { ns: namespace });
  lang.SEARCH = i18n.t(`Search`, { ns: namespace });
}

// run it initial
fill();

// bind some events and fill values again (doing the magic you expect to happen magically)
i18n.on(`languageChanged`, () => {
  fill();
});

// export the const
export default lang;
