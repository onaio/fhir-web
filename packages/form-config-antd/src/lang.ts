import i18n, { namespace } from './mls';
import { Dictionary } from '@onaio/utils';

export type Lang = Dictionary<string>;
const lang: Lang = {};

/** recompute values */
function fill() {
  // Errors
  lang.ERROR_FORM_REQUIRED = i18n.t(`Form is required`, { ns: namespace });
  lang.ERROR_FORM_NAME_REQUIRED = i18n.t(`Form Name is required`, { ns: namespace });

  lang.MAKE_RELEASE = i18n.t(`Make Release`, { ns: namespace });
  lang.FILE_NAME = i18n.t(`File Name`, { ns: namespace });
  lang.FILE_VERSION = i18n.t(`File Version`, { ns: namespace });
  lang.IDENTIFIER = i18n.t(`Identifier`, { ns: namespace });
  lang.MODULE = i18n.t(`Module`, { ns: namespace });
  lang.EDIT = i18n.t(`Edit`, { ns: namespace });
  lang.DOWNLOAD = i18n.t(`Download`, { ns: namespace });
  lang.UPLOAD_EDIT = i18n.t(`Upload edit`, { ns: namespace });
  lang.UPLOAD_NEW_FILE = i18n.t(`Upload New File`, { ns: namespace });
  lang.VIEW_FILES = i18n.t(`View Files`, { ns: namespace });
  lang.APP_ID_LABEL = i18n.t(`App ID`, { ns: namespace });
  lang.APP_VERSION_LABEL = i18n.t(`App Version`, { ns: namespace });
  lang.RELATED_TO = i18n.t(`Related to`, { ns: namespace });
  lang.UPLOAD_FILE = i18n.t(`Upload file`, { ns: namespace });
  lang.FIND_DRAFT_FILES = i18n.t(`Find Draft Files`, { ns: namespace });
  lang.FIND_FILES = i18n.t(`Find Files`, { ns: namespace });
  lang.FIND_RELEASES_LABEL = i18n.t(`Find Release`, { ns: namespace });
  lang.SEARCH = i18n.t(`Search`, { ns: namespace });
  lang.CREATED_AT = i18n.t(`Created at`, { ns: namespace });
  lang.UPDATED_AT_LABEL = i18n.t(`Updated at`, { ns: namespace });
  lang.ERROR_OCCURRED = i18n.t(`Error occurred`, { ns: namespace });
  lang.DRAFT_FILES = i18n.t(`Draft Files`, { ns: namespace });
  lang.RELEASES = i18n.t(`Releases`, { ns: namespace });
  lang.JSON_VALIDATORS = i18n.t(`JSON Validators`, { ns: namespace });
  lang.ACTION = i18n.t(`Action`, { ns: namespace });
  lang.UPLOAD_FORM = i18n.t(`Upload Form`, { ns: namespace });
  lang.FORM_NAME = i18n.t(`Form Name`, { ns: namespace });
  lang.FORM = i18n.t(`Form`, { ns: namespace });
  lang.UPLOADING = i18n.t(`Uploading`, { ns: namespace });
  lang.CLICK_TO_UPLOAD = i18n.t(`Click to upload`, { ns: namespace });
}

// run it initial
fill();

// bind some events and fill values again (doing the magic you expect to happen magically)
i18n.on(`languageChanged`, () => {
  fill();
});

// export the const
export default lang;
