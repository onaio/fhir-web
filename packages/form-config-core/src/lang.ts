import i18n from './mls';

export const ERROR_OCCURRED = i18n.t('Error occurred');
