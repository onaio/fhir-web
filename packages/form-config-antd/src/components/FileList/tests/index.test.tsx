/* eslint-disable @typescript-eslint/camelcase */
import React from 'react';
import { mount, shallow } from 'enzyme';
import { createBrowserHistory } from 'history';
import { FileList } from '../index';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import { store } from '@opensrp/store';
import reducerRegistry from '@onaio/redux-reducer-registry';
import flushPromises from 'flush-promises';
import fetch from 'jest-fetch-mock';
import { filesReducer, filesReducerName, removeManifestFiles } from '@opensrp/form-config-core';
import {
  fixManifestFiles,
  downloadFile,
  fixManifestReleases,
  manifestFile1,
  manifestFile2,
  manifestFile3,
} from '../../../helpers/fixtures';
import * as helpers from '@opensrp/form-config-core';
import { act } from 'react-dom/test-utils';
import { authenticateUser } from '@onaio/session-reducer';
import * as notifications from '@opensrp/notifications';
import lang from '../../../lang';

jest.mock('@opensrp/notifications', () => ({
  __esModule: true,
  ...Object.assign({}, jest.requireActual('@opensrp/notifications')),
}));

jest.mock('@opensrp/form-config-core', () => ({
  __esModule: true,
  ...Object.assign({}, jest.requireActual('@opensrp/form-config-core')),
}));

const mockNotificationError = jest.spyOn(notifications, 'sendErrorNotification');

/** register the reducers */
reducerRegistry.register(filesReducerName, filesReducer);

const history = createBrowserHistory();
const mockHistoryPush = jest.fn();
history.push = mockHistoryPush;

describe('components/Antd/FileList', () => {
  beforeAll(() => {
    store.dispatch(
      authenticateUser(
        true,
        {
          email: 'bob@example.com',
          name: 'Bobbie',
          username: 'RobertBaratheon',
        },
        // eslint-disable-next-line @typescript-eslint/camelcase
        { api_token: 'hunter2', oAuth2Data: { access_token: 'hunter2', state: 'abcde' } }
      )
    );
  });

  afterEach(() => {
    store.dispatch(removeManifestFiles());
    jest.clearAllMocks();
    fetch.resetMocks();
  });

  const props = {
    opensrpBaseURL: 'https://test-example.com/rest/',
    uploadFileURL: '/form-upload',
    isJsonValidator: true,
    history,
    location: {
      hash: '',
      pathname: `/json-validators`,
      search: '',
      state: {},
    },
    match: {
      isExact: true,
      params: { formVersion: null },
      path: '/json-validators',
      url: `/json-validators`,
    },
  };

  it('renders without crashing', () => {
    shallow(
      <Provider store={store}>
        <Router history={history}>
          <FileList {...props} />
        </Router>
      </Provider>
    );
  });

  it('renders correctly', async () => {
    fetch.once(JSON.stringify(fixManifestFiles));
    fetch.once(JSON.stringify(downloadFile));

    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <FileList {...props} />
        </Router>
      </Provider>
    );

    await act(async () => {
      await flushPromises();
    });

    wrapper.update();

    expect(fetch.mock.calls[0]).toEqual([
      `https://test-example.com/rest/clientForm/metadata?is_json_validator=true`,
      {
        headers: {
          accept: 'application/json',
          authorization: 'Bearer hunter2',
          'content-type': 'application/json;charset=UTF-8',
        },
        method: 'GET',
      },
    ]);
    const content = wrapper.find('div.layout-content');
    expect(content.find('Title').props()).toMatchSnapshot('title');
    expect(content.find('Card').prop('children')).toHaveLength(3);
    expect(content.find('Card').prop('children')[0]).toMatchSnapshot('search');
    expect(content.find('Card').prop('children')[1]).toMatchSnapshot('upload button');
    expect(content.find('Card').prop('children')[2]).toMatchSnapshot('table');
    wrapper.unmount();
  });

  it('renders correctly for view file', async () => {
    fetch.once(JSON.stringify(fixManifestFiles));
    fetch.once(JSON.stringify(downloadFile));

    const viewFilesProps = {
      ...props,
      uploadFileURL: '',
      isJsonValidator: false,
      history,
      location: {
        hash: '',
        pathname: `/releases/${fixManifestReleases[0].identifier}`,
        search: '',
        state: {},
      },
      match: {
        isExact: true,
        params: { formVersion: fixManifestReleases[0].identifier },
        path: '/releases:formVersion',
        url: `/releases/${fixManifestReleases[0].identifier}`,
      },
    };

    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <FileList {...viewFilesProps} />
        </Router>
      </Provider>
    );

    await act(async () => {
      await flushPromises();
    });

    wrapper.update();

    expect(fetch.mock.calls[0]).toEqual([
      `https://test-example.com/rest/clientForm/metadata?identifier=${fixManifestReleases[0].identifier}`,
      {
        headers: {
          accept: 'application/json',
          authorization: 'Bearer hunter2',
          'content-type': 'application/json;charset=UTF-8',
        },
        method: 'GET',
      },
    ]);
    const content = wrapper.find('div.layout-content');
    expect(content.find('Title').props()).toMatchSnapshot('title');
    expect(content.find('Card').prop('children')).toHaveLength(3);
    expect(content.find('Card').prop('children')[0]).toMatchSnapshot('search');
    expect(content.find('Card').prop('children')[1]).toMatchSnapshot('upload button');
    expect(content.find('Card').prop('children')[2]).toMatchSnapshot('table');
    wrapper.unmount();
  });

  it('searches correctly', async () => {
    fetch.once(JSON.stringify([manifestFile1, manifestFile2, manifestFile3]));
    fetch.once(JSON.stringify(downloadFile));

    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <FileList {...props} />
        </Router>
      </Provider>
    );

    await act(async () => {
      await flushPromises();
    });
    wrapper.update();

    const search = wrapper.find('input#search');
    // Before search
    expect(wrapper.find('tbody').find('tr')).toHaveLength(3);

    // Search by label
    search.simulate('change', { target: { value: 'test publish' } });
    wrapper.update();
    expect(wrapper.find('tbody').find('tr')).toHaveLength(1);

    // Reset
    search.simulate('change', { target: { value: '' } });
    wrapper.update();
    expect(wrapper.find('tbody').find('tr')).toHaveLength(3);

    // Search by identifier
    search.simulate('change', { target: { value: 'test-form' } });
    wrapper.update();
    expect(wrapper.find('tbody').find('tr')).toHaveLength(1);

    // Reset
    search.simulate('change', { target: { value: '' } });
    wrapper.update();
    expect(wrapper.find('tbody').find('tr')).toHaveLength(3);

    // Search by module
    search.simulate('change', { target: { value: 'baz' } });
    wrapper.update();
    expect(wrapper.find('tbody').find('tr')).toHaveLength(1);

    wrapper.unmount();
  });

  it('download files correctly', async () => {
    fetch.once(JSON.stringify(fixManifestFiles));
    fetch.once(JSON.stringify(downloadFile));
    const downloadSpy = jest.spyOn(helpers, 'downloadManifestFile');

    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <FileList {...props} />
        </Router>
      </Provider>
    );

    await act(async () => {
      await flushPromises();
    });

    wrapper.update();

    const downloadCell = wrapper.find('tbody').find('tr').at(0).find('td').at(4);
    const dropdown = downloadCell.find('Dropdown');
    const submenu = shallow(<div>{dropdown.prop('overlay')}</div>);

    expect(submenu.find('Button').text()).toEqual('Download');

    // click download button
    submenu.find('Button').simulate('click');
    await act(async () => {
      await flushPromises();
    });

    wrapper.update();

    expect(downloadSpy).toHaveBeenCalledWith(
      'hunter2',
      'https://test-example.com/rest/',
      'clientForm',
      {
        createdAt: 'Jun 19, 2020, 12:31:37 PM',
        form_relation: '',
        id: '52',
        identifier: 'test-form-1.json',
        isDraft: false,
        isJsonValidator: false,
        jursdiction: '',
        key: 0,
        label: 'test form',
        module: '',
        version: '1.0.26',
      },
      true,
      undefined
    );
    expect(fetch.mock.calls[1]).toEqual([
      `https://test-example.com/rest/clientForm?form_identifier=${fixManifestFiles[0].identifier}&form_version=${fixManifestFiles[0].version}&is_json_validator=true`,
      {
        headers: {
          accept: 'application/json',
          authorization: 'Bearer hunter2',
          'content-type': 'application/json;charset=UTF-8',
        },
        method: 'GET',
      },
    ]);
    wrapper.unmount();
  });

  it('download with correct params form normal file', async () => {
    const downloadSpy = jest.spyOn(helpers, 'downloadManifestFile');
    fetch.once(JSON.stringify(fixManifestFiles));
    fetch.once(JSON.stringify(downloadFile));
    // file not json validator
    const normalFileProps = {
      ...props,
      isJsonValidator: false,
    };

    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <FileList {...normalFileProps} />
        </Router>
      </Provider>
    );
    await act(async () => {
      await flushPromises();
    });

    wrapper.update();

    const downloadCell = wrapper.find('tbody').find('tr').at(0).find('td').at(5);
    const dropdown = downloadCell.find('Dropdown');
    const submenu = shallow(<div>{dropdown.prop('overlay')}</div>);

    expect(submenu.find('Button').text()).toEqual('Download');

    // click download button
    submenu.find('Button').simulate('click');
    await act(async () => {
      await flushPromises();
    });
    wrapper.update();
    expect(downloadSpy).toHaveBeenCalledWith(
      'hunter2',
      'https://test-example.com/rest/',
      'clientForm',
      {
        createdAt: 'Jun 19, 2020, 12:31:37 PM',
        form_relation: '',
        id: '52',
        identifier: 'test-form-1.json',
        isDraft: false,
        isJsonValidator: false,
        jursdiction: '',
        key: 0,
        label: 'test form',
        module: '',
        version: '1.0.26',
      },
      false,
      undefined
    );
    expect(fetch.mock.calls[1]).toEqual([
      `https://test-example.com/rest/clientForm?form_identifier=${fixManifestFiles[0].identifier}&form_version=${fixManifestFiles[0].version}`,
      {
        headers: {
          accept: 'application/json',
          authorization: 'Bearer hunter2',
          'content-type': 'application/json;charset=UTF-8',
        },
        method: 'GET',
      },
    ]);
    wrapper.unmount();
  });

  it('handles failure if fetching mainfest files fails', async () => {
    fetch.mockRejectOnce(() => Promise.reject('API is down'));
    fetch.once(JSON.stringify(downloadFile));

    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <FileList {...props} />
        </Router>
      </Provider>
    );

    await act(async () => {
      await flushPromises();
    });
    wrapper.update();

    expect(mockNotificationError).toHaveBeenCalledWith(lang.ERROR_OCCURRED);
    expect(wrapper.find('tbody').find('tr').find('td').find('div.ant-empty-image')).toHaveLength(1);

    wrapper.unmount();
  });

  it('handles download file failure', async () => {
    fetch.once(JSON.stringify(fixManifestFiles));
    fetch.mockRejectOnce(() => Promise.reject('Cannot fetch file'));
    const downloadSpy = jest.spyOn(helpers, 'handleDownload');

    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <FileList {...props} />
        </Router>
      </Provider>
    );

    await act(async () => {
      await flushPromises();
    });
    wrapper.update();

    expect(wrapper.find('tbody').find('tr')).toHaveLength(fixManifestFiles.length);

    const downloadFiledCell = wrapper.find('tbody').find('tr').at(0).find('td').at(4);
    const dropdown = downloadFiledCell.find('Dropdown');
    const submenu = shallow(<div>{dropdown.prop('overlay')}</div>);

    submenu.find('Button').simulate('click');

    await act(async () => {
      await flushPromises();
    });

    wrapper.update();
    expect(mockNotificationError).toHaveBeenCalledWith(lang.ERROR_OCCURRED);
    expect(downloadSpy).not.toHaveBeenCalled();
    wrapper.unmount();
  });

  it('upload file button works', async () => {
    fetch.once(JSON.stringify(fixManifestFiles));

    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <FileList {...props} />
        </Router>
      </Provider>
    );

    await act(async () => {
      await flushPromises();
    });
    wrapper.update();

    wrapper.find('Space').at(1).find('button').simulate('click');

    expect(mockHistoryPush).toHaveBeenCalledWith('/form-upload');

    wrapper.unmount();
  });
});
