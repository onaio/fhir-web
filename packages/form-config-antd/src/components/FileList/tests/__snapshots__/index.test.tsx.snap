// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Antd/FileList renders correctly for view file: search 1`] = `
<Space
  style={
    Object {
      "float": "left",
      "marginBottom": 16,
    }
  }
>
  <Input
    id="search"
    onChange={[Function]}
    placeholder="Search"
    prefix={<ForwardRef(SearchOutlined) />}
    size="large"
    type="text"
    value=""
  />
</Space>
`;

exports[`components/Antd/FileList renders correctly for view file: table 1`] = `
<TableLayout
  actions={
    Object {
      "render": [Function],
    }
  }
  columns={
    Array [
      Object {
        "dataIndex": "identifier",
        "ellipsis": true,
        "key": "identifier",
        "sortOrder": undefined,
        "sorter": [Function],
        "title": "Identifier",
      },
      Object {
        "dataIndex": "label",
        "ellipsis": true,
        "key": "label",
        "sortOrder": undefined,
        "sorter": [Function],
        "title": "File Name",
      },
      Object {
        "dataIndex": "version",
        "ellipsis": true,
        "key": "version",
        "sortOrder": undefined,
        "sorter": [Function],
        "title": "File Version",
      },
      Object {
        "dataIndex": "createdAt",
        "ellipsis": true,
        "key": "createdAt",
        "render": [Function],
        "sortOrder": undefined,
        "sorter": [Function],
        "title": "Created at",
      },
      Object {
        "dataIndex": "module",
        "key": "module",
        "title": "Module",
      },
      Object {
        "key": "_actions",
        "render": [Function],
        "title": "Actions",
      },
    ]
  }
  datasource={
    Array [
      Object {
        "createdAt": "Jun 19, 2020, 12:31:37 PM",
        "form_relation": "",
        "id": "52",
        "identifier": "test-form-1.json",
        "isDraft": false,
        "isJsonValidator": false,
        "jursdiction": "",
        "label": "test form",
        "module": "",
        "version": "1.0.26",
      },
      Object {
        "createdAt": "Jun 19, 2020, 4:23:22 PM",
        "form_relation": "",
        "id": "53",
        "identifier": "reveal-test-file.json",
        "isDraft": false,
        "isJsonValidator": false,
        "jursdiction": "",
        "label": "test publish",
        "module": "",
        "version": "1.0.27",
      },
    ]
  }
  id="FormFileList"
  onChange={[Function]}
  persistState={true}
/>
`;

exports[`components/Antd/FileList renders correctly for view file: title 1`] = `
Object {
  "children": "Releases: 1.0.11",
  "level": 3,
}
`;

exports[`components/Antd/FileList renders correctly for view file: upload button 1`] = `
<Space
  style={
    Object {
      "float": "right",
      "marginBottom": 16,
    }
  }
>
  <ForwardRef(SettingOutlined) />
</Space>
`;

exports[`components/Antd/FileList renders correctly: search 1`] = `
<Space
  style={
    Object {
      "float": "left",
      "marginBottom": 16,
    }
  }
>
  <Input
    id="search"
    onChange={[Function]}
    placeholder="Search"
    prefix={<ForwardRef(SearchOutlined) />}
    size="large"
    type="text"
    value=""
  />
</Space>
`;

exports[`components/Antd/FileList renders correctly: table 1`] = `
<TableLayout
  actions={
    Object {
      "render": [Function],
    }
  }
  columns={
    Array [
      Object {
        "dataIndex": "identifier",
        "ellipsis": true,
        "key": "identifier",
        "sortOrder": undefined,
        "sorter": [Function],
        "title": "Identifier",
      },
      Object {
        "dataIndex": "label",
        "ellipsis": true,
        "key": "label",
        "sortOrder": undefined,
        "sorter": [Function],
        "title": "File Name",
      },
      Object {
        "dataIndex": "version",
        "ellipsis": true,
        "key": "version",
        "sortOrder": undefined,
        "sorter": [Function],
        "title": "File Version",
      },
      Object {
        "dataIndex": "createdAt",
        "ellipsis": true,
        "key": "createdAt",
        "render": [Function],
        "sortOrder": undefined,
        "sorter": [Function],
        "title": "Created at",
      },
      Object {
        "key": "_actions",
        "render": [Function],
        "title": "Actions",
      },
    ]
  }
  datasource={
    Array [
      Object {
        "createdAt": "Jun 19, 2020, 12:31:37 PM",
        "form_relation": "",
        "id": "52",
        "identifier": "test-form-1.json",
        "isDraft": false,
        "isJsonValidator": false,
        "jursdiction": "",
        "label": "test form",
        "module": "",
        "version": "1.0.26",
      },
      Object {
        "createdAt": "Jun 19, 2020, 4:23:22 PM",
        "form_relation": "",
        "id": "53",
        "identifier": "reveal-test-file.json",
        "isDraft": false,
        "isJsonValidator": false,
        "jursdiction": "",
        "label": "test publish",
        "module": "",
        "version": "1.0.27",
      },
    ]
  }
  id="FormFileList"
  onChange={[Function]}
  persistState={true}
/>
`;

exports[`components/Antd/FileList renders correctly: title 1`] = `
Object {
  "children": "JSON Validators",
  "level": 3,
}
`;

exports[`components/Antd/FileList renders correctly: upload button 1`] = `
<Space
  style={
    Object {
      "float": "right",
      "marginBottom": 16,
    }
  }
>
  <React.Fragment>
    <Button
      block={false}
      ghost={false}
      htmlType="button"
      loading={false}
      onClick={[Function]}
      type="primary"
    >
      <ForwardRef(UploadOutlined) />
      Upload New File
    </Button>
    <Divider
      type="vertical"
    />
  </React.Fragment>
  <ForwardRef(SettingOutlined) />
</Space>
`;
